exchangelib/account.py

```diff
@@ -747,6 +756,69 @@ def delegates(self):
         """Return a list of DelegateUser objects representing the delegates that are set on this account."""
         return list(GetDelegate(account=self).call(user_ids=None, include_permissions=True))
 
+    @property
+    def rules(self):
+        """Return a list of Rule objects representing the rules that are set on this account."""
+        try:
+            elems = GetInboxRules(account=self).call()
+            return list(elems)
+        except MalformedResponseError as exc:
+            empty_info = f"No {{{MNS}}}InboxRules elements in ResponseMessage"
+            if empty_info in str(exc):
+                return []
+            raise exc
+
+    def get_rules(self, generator=False) -> List[Rule]:
+        """Retrieve Inbox rules in the identified user's mailbox.
+
+        :param generator: If True, return a generator instead of a list. Default is False.
+        :return: A list of Rule objects if generator is False, else a generator.
+        """
+        if generator:
+            # If the generator parameter is True, return a generator.
+            return GetInboxRules(account=self).call()
+        return self.rules
+
+    def create_rule(self, rule: Rule):
+        """Create an Inbox rule in a user's mailbox in the Exchange store.
+
+        :param rule: The rule to create with display_name as the key at least.
+        :return: The rule ID of the created rule. If failed, raise an error.
+        """
+        create_rule_service = CreateInboxRule(account=self)
+        create_rule_service.get(rule=rule, remove_outlook_rule_blob=True)
+        # After creating the rule, query all rules,
+        # find the rule that was just created, and return its ID.
```

Comment:
Doesn't the `CreateInboxRule` service return an ID of the created rule? All other services do that.

Reply:
According to https://learn.microsoft.com/en-us/exchange/client-developer/web-service-reference/updateinboxrules-operation#code-1, CreateInBoxrule Services does not return ID.

Reply:
Ah, thanks for the pointer to the docs. I'm surprised - EWS is usually better designed than this.